File: ios/LookSeeProto/Models/ArchivedMedia.swift
//
//  ArchivedMedia.swift
//  LookSeeProto
//
//  Created by Angel Pineda on 6/18/26.
//

import Foundation

struct ArchivedMedia: Identifiable, Codable {
    var id = UUID()
    var title: String
    var fileName: String
    var thumbnailFileName: String
    var isVideo: Bool
    var latitude: Double
    var longitude: Double
    var dateSaved: Date
    var isFavorite: Bool?
    
    // --- NEW QUEUE FIELDS ---
    var landmarkId: String?
    var savedLabel: String?
    var savedDescription: String?
    var savedUserDescription: String?
    var negativeVideoFileName: String?
    
    // Differentiates between Tier-1 (Record) and Tier-2 (Uploads)
    var isTier2: Bool?
}

File: Android/LookSee/app/src/main/java/looksee/angelll/com/models/ArchivedMedia.kt
package looksee.angelll.com.models

import java.util.Date
import java.util.UUID

/** Media saved locally for the archive and offline-upload queue. */
data class ArchivedMedia(
    val id: UUID = UUID.randomUUID(),
    val title: String,
    val fileName: String,
    val thumbnailFileName: String,
    val isVideo: Boolean,
    val latitude: Double,
    val longitude: Double,
    val dateSaved: Date,
    val isFavorite: Boolean? = null,
    val landmarkId: String? = null,
    val savedLabel: String? = null,
    val savedDescription: String? = null,
    val savedUserDescription: String? = null,
    val negativeVideoFileName: String? = null,
    val isTier2: Boolean? = null,
)
